src/main/java/view/SearchView.java

```diff
@@ -112,6 +116,46 @@ public void actionPerformed(ActionEvent e) {
                 SearchView.this.addFilter("Cast", castField.getText());
             }
         });
+
+        // Action listener for keyword search button (search by name)
+        keywordSearchButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final String keyword = keywordField.getText().trim();
+
+                if (keyword.isEmpty()) {
+                    showError("Please enter a movie name to search.");
+                }
+                else {
+                    clearError();
+                    searchController.execute(keyword);
+                }
+            }
+        });
+
+        // Action listener for main search button
+        mainSearchButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final String keyword = keywordField.getText().trim();
+                final List<String> genres = new ArrayList<>();
+                if (!genreField.getText().trim().isEmpty()) {
+                    genres.add(genreField.getText().trim());
+                }
+                final List<String> cast = new ArrayList<>();
+                if (!castField.getText().trim().isEmpty()) {
+                    cast.add(castField.getText().trim());
+                }
+
+                if (keyword.isEmpty() && genres.isEmpty() && cast.isEmpty()) {
+                    showError("At least one filter (Keyword, Genre, Cast) is required.");
+                }
+                else {
+                    clearError();
+                    searchViewModel.performSearch(keyword, genres, cast);
```

Comment:
This should be passed on to the search controller, not a view model. If you think this cannot (or shouldn't, if it violates the single responsibility principle) be handled by your existing controller/interactor/presenter, then Clean Architecture dictates that you should make new ones. If you go ahead with it, you can vastly reduce the amount of code to write by copying over your current Search classes and using them as templates.